### DISCUSSION TITLE
The CFL may have abandoned their public API.

### DISCUSSION THREAD

**Discussion by mdryden** [110yards/fantasy#234] [Announcements]
**TL;DR:** It appears to me that the CFL has abandoned their API, and if that is the case, 110 yards will not get any stat updates when games start tomorrow.

### Why do I think this?

1) At some point during the offseason, the CFL let their SSL certificate for the API expire.  I was able to work around this by using the http endpoint, but the fact that no one is on top of a basic infrastructure task like this is concerning.

2) None of the pre-season games received any stat updates.  No scores, no player stats, they don't even show that the games themselves have started.  This fact can also be seen on the pre-season gametracker pages on cfl.ca, which also didn't get updated: https://www.cfl.ca/games/6302/toronto-argonauts-vs-ottawa-redblacks/

3) Multiple players that I know of have bad data in the API.  Taquan Mizzell, for instance is listed as a wide receiver and not a running back.

### What happens next?

I don't know yet, but I will share updates here as I get them.  I have been in contact with someone who has connections within the CFL  so hopefully I will get more information on what's happening.

My goal is to keep 110 yards alive, but I think there is a very good chance that 1 or more week will be affected if the API is truly dead.

For the time being, I have disabled the donation link to ensure I don't accept any money if the site does need to be shutdown for the season.

### June 8th update - 12 PM Central

It's highly likely that the site will not get stat updates today, but until that is 100% certain, I'm not going to make any changes to the site.

I do not intend to abandon the site and am certain I can find some option for stats, though there is a chance that we either may lose the ability to score some stat categories, depending on what I land on as a source.  I'm investigating multiple options, including Genius Sports.

Please don't give up on your leagues if the stats don't end up calculating after game 1 this week, I am still hoping to salvage something in time to have scoring available before the end of week 1.  I'll continue to provide updates here as I get more information.

### June 8th update 2 - 9PM Central

Well the game has started and the worst has come to fruition - there are no live stats.

For today I've manually locked the teams which are playing so if we do find a solution, managers can't move players now that the games are in progress, but I won't be able to do this for the whole weekend.  I may be able to hack in a quick fix to lock players at the start of the game automatically.

The bad news is that I'm not seeing any good paths forward for actual stats, especially not on short notice.  The CFL's new game tracker does not provide any player IDs, and even if I pulled stats from there, there's no way to connect those stats to a player on the site without doing a name match, which is not something I trust (I've done this in the past for cflstats.ca and it was a nightmare).

I'm not sure what my next step is going to be, but I'm still thinking on it.

### June 8th update 3 - 10:30 PM Central

I believe I have a path forward.  Please be patient while I investigate.  My goal is to have stats calculated for week 1 prior to the start of week 2, but if that does not happen, I will do my best to recalculate any missed weeks once the stats are ready.

### June 9th update - 12 AM Central

Made significant progress on some parts of the fix, but ran into a major roadblock on another.  I have to call it for the night and unfortunately I'm not available to do any work on this again until Sunday night due to family commitments.

I remain hopeful that I can have a fix in place early next week (in time for week 2).

### June 12 update 6:40 pm central

I made significant progress on an alternate source of stats on the weekend and have more worked planned for tonight.

My source near the CFL also informs me that there may be a desire within the CFL to keep the API active after all; I am hoping to get an update on that tomorrow evening.

I have two plans currently:

1) finish my backup option and have it ready in the next day or two

2) wait for the CFL to bring the API back online.

The update I get tomorrow may dictate the direction.

Option 1 will work but comes with risk and losses of data so option 2 is my preference.

Waivers will not be ready to run tomorrow morning, but I remain optimistic that the stats will be in place in time to score this week before next week starts.

### Update June 13 - 4PM Central

The death of the CFL's API has now been officially-unofficially confirmed so I will be proceeding with option 1 tonight.

My goal is to have scores calculated and waivers activated by tomorrow morning.

**Comment by kamikazecanuck** [comment]
NOOOOO!  I've noticed that even on CFL.ca they don' have any preseason stats.  I find it hard to imagine that they won't have real time stats for the real games.  Is it possible all will be ok once the first kickoff happens?

**Reply by mdryden** [reply]
It's possible, I just want to get out ahead of it in case it goes sideways.

If the API starts updating tomorrow then this will all be much ado about nothing.  If it doesn't, then we're probably screwed for week 1.  From there, it will depend on what my contact says about the API, I'll either wait for a fix, or I'll see if I can work out another source of stats.

The problem is that the only source of stats that I know of is Stats, Inc, and their prices are astronomical.

In the past I was able to parse the stats for cflstats.ca out of published box scores; that may be an option going forward, but it will take some time to get something in place if I have to build my own API.

**Reply by kamikazecanuck** [reply]
OK, most of this is beyond me but although some of the player info is not totally accurate like some people being labelled a free agent when they're not it is overall accurate in that it removed players who've left the league and added ones that have joined.  Hopefully that's a good sign?

Does the cflstats.ca option require a lot of manual data entry?

**Reply by mdryden** [reply]
It did previously, yes.., but only because I used to manually correct errors in the play by play; the stats themselves were pretty reliable.

It's not likely something I can whip out in a day though, so I'm going to wait and see what happens tomorrow.

**Reply by MBControl-yyc** [reply]
I really appreciate the heads up.  You run a great site.

1)  My GM's and I are willing to pay something of a yearly fee.  We do donate, but we would be okay with a mandatory fee.  Not Ideal, and probably doesn't make a dent, but just want you to know we value you and the site.

2)  In the time between Fantaseh closing down, and your site beginning, I ran our league using Google Sheets, entering stats every Monday night.  What I am saying is.  If you have a solution that involves data entry and could use some help, I will volunteer myself for the season.

3)  If there is anything from the community that could be of help to you, please don't hesitate to ask us for it.  You're doing the heavy lifting here, but I'm sure we would like to chip in if we can.

**Reply by argos1873** [reply]
Does this have anything to do with the league's transition to Genius Sports' LiveStats system?

https://3downnation.com/2023/05/18/cfl-genius-sports-unveil-new-live-stats-system-tailored-for-betting-data/

**Reply by argos1873** [reply]
Also, seconding that I'm willing to volunteer my time to help this season run smoothly, if need be

**Reply by davidkinetisense** [reply]
It appears you are correct.

Their Fantasy sucks, I just looked at it.

**Reply by kamikazecanuck** [reply]
> Does this have anything to do with the league's transition to Genius Sports' LiveStats system?
> 
> https://3downnation.com/2023/05/18/cfl-genius-sports-unveil-new-live-stats-system-tailored-for-betting-data/

I’m sure it does. Thanks Ambroise for selling 10% of the league to some chumps who no one cares about and don’t care about the league.

**Reply by mdryden** [reply]
> I really appreciate the heads up. You run a great site.
> 
> 1. My GM's and I are willing to pay something of a yearly fee.  We do donate, but we would be okay with a mandatory fee.  Not Ideal, and probably doesn't make a dent, but just want you to know we value you and the site.
> 2. In the time between Fantaseh closing down, and your site beginning, I ran our league using Google Sheets, entering stats every Monday night.  What I am saying is.  If you have a solution that involves data entry and could use some help, I will volunteer myself for the season.
> 3. If there is anything from the community that could be of help to you, please don't hesitate to ask us for it.  You're doing the heavy lifting here, but I'm sure we would like to chip in if we can.

Thank you, I appreciate all of that; this community means a lot to me.

Right now there's nothing to be done, we just have to wait and see.  As mentioned in the update above, I think there is a path forward, but my options are limited until the season actually starts or until I get a response from other stat providers.

**Reply by kamikazecanuck** [reply]
Thanks again.  My league is actually considering going back the excel days (dark, dark days).  But using 110yards to govern the rosters.  So hopefully you keep the site up to at least deal with that.  Roster moves and waiver wires is like 50% of the work right there.  Hopefully, there's a last minute Hail Mary in here somewhere...

**Reply by mdryden** [reply]
I'll keep the site up for that.

**Comment by davidkinetisense** [comment]
FROM CFL TECH SUPPORT "Hi David –

Thanks for reaching out. The league is currently overhauling the entire stats system including CFL Connect. Going forward, only select personnel will have access to it.

For fans, Gametracker will be ready to go for kickoff and you can follow the stats live.  I hope this helps. Happy to answer any further questions you may have.

Enjoy the season kickoff tomorrow night!

Lucas Barrett" So dumb.

I have successfully used Octoparse in between Fantash and 110 yards to scrape the data from the website.  However, if that is not updated weekly on a website we would not be able to access it.  This may be a solution so we have the data loaded weekly.  I am attaching what I scraped now from the CFL player's page.

[Players - CFL.ca.csv](https://github.com/mdryden/110yards/files/11682549/Players.-.CFL.ca.csv)

**Reply by davidkinetisense** [reply]
Michael, what tables would you need?  Would scraping the data be an option?  We would not know who the winners are until the end of the weekend.

Thanks, David

**Reply by streibeb** [reply]
I've sent them two emails to no response over the last week. How did you get one so quick!?

**Comment by davidkinetisense** [comment]
The $70,000 dollar weekly bidding sucks.  We want a 110yards.ca fantasy
league.  Sell your system to the CFL. :)  We want this type of fantasy.

Thanks, David

On Wed, Jun 7, 2023 at 6:26 PM kamikazecanuck ***@***.***>
wrote:

> Does this have anything to do with the league's transition to Genius
> Sports' LiveStats system?
>
>
> https://3downnation.com/2023/05/18/cfl-genius-sports-unveil-new-live-stats-system-tailored-for-betting-data/
>
> I’m sure it does. Thanks Ambroise for selling 10% of the league to some
> chumps who no one cares about and don’t care about the league.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mdryden/110yards/discussions/234#discussioncomment-6116908>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APOMMCRENCVTJLK4RH742Z3XKEL5BANCNFSM6AAAAAAY6DRTP4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*David Schnare*
Kinetisense Inc. Co-founder
403-526-3556

**Comment by davidkinetisense** [comment]
How many leagues are there?

**Comment by davidkinetisense** [comment]
Not sure.  Sent it to tech@

**Reply by streibeb** [reply]
That's exactly where I sent mine too.  Zero response after two weeks.  Tried using the Contact Us form on the main site as well, zero results.

**Comment by kamikazecanuck** [comment]
Well it's game day...Is it pretty much confirmed 110yards won't work today?

**Comment by davidkinetisense** [comment]
Do we know how many people use 110yards

**Comment by mdryden** [comment]
There are 692 registered users on the site, and 61 active leagues this year.

**Reply by kamikazecanuck** [reply]
Wow, didn't know it was that many.  CFL should really take that as a hint.

**Comment by davidkinetisense** [comment]
Is scraping the data an option.   I am willing to help with this.  David

On Thu, Jun 8, 2023 at 11:05 AM Michael Dryden ***@***.***>
wrote:

> There are 692 registered users on the site, and 61 active leagues this
> year.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mdryden/110yards/discussions/234#discussioncomment-6124934>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APOMMCQ26Z7OOMGDVAY25IDXKIA4ZANCNFSM6AAAAAAY6DRTP4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
*David Schnare*
Kinetisense Inc. Co-founder
403-526-3556

**Reply by mdryden** [reply]
Scraping is one option I'm considering, however the league is probably aware of this site (I was a registered user of the API), and therefore I would prefer an option which isn't in such a legal grey area.

**Reply by kamikazecanuck** [reply]
I think the fact we're in this situation shows that the league doesn't really care what's going on out there and also doesn't have the capability to know things.

**Comment by davidkinetisense** [comment]
The CFL does not offer anything thing like your site.  Their type of
Fantasy is not fun.

Well, let us know if there is anything we can do to help you.  There are a
lot of disappointing people at this point. :(

David

On Thu, Jun 8, 2023 at 11:11 AM Michael Dryden ***@***.***>
wrote:

> Scraping is one option I'm considering, however the league is probably
> aware of this site (I was a registered user of the API), and therefore I
> would prefer an option which isn't in such a legal grey area.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mdryden/110yards/discussions/234#discussioncomment-6124988>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APOMMCUZWPMIUTBUNGIY4CTXKIBULANCNFSM6AAAAAAY6DRTP4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*David Schnare*
Kinetisense Inc. Co-founder
403-526-3556

**Reply by kamikazecanuck** [reply]
Agree, Weekly Player selector is so dumb.  I don't know why they can't have real fantasy.  That's what gets people invested in the players imo.  

There's a few volunteers....maybe each volunteer could be assigned data entry for each team? I don't know, just spitballing here.

**Reply by streibeb** [reply]
The league fantasy does have head-to-head options now, but no season-long fantasy.  Step in the right direction.

**Reply by JonathanClink** [reply]
Head to head doesn't translate to gambling as much as draft kings-style games. It's strange they neglect head to head when it's so popular for NFL.

**Comment by MBControl-yyc** [comment]
It is a lot of people, but probably not enough to make the league take too much notice.  With that said, my league has taken 8-10 people that might watch their favorite team play each week, now watch 99% of all CFL games broadcast (expect for Montreal home games.  Horn guy sucks).  They know the 2nd string back DB's for Ottawa, watching CFL news and rumors.

This type of fantasy encourages a much higher level of engagement to the league.  But as much as I love CFL football.  The league is a very poorly run league.

**Reply by kamikazecanuck** [reply]
Yep that pretty much sums it up.  Been running a keeper league for like 10 years.  People who where not even CFL fans have come to really know the trails and tribulations of their players.  That's something the CFL needs to stamp out immediately.

**Comment by davidkinetisense** [comment]
Well said.

On Thu, Jun 8, 2023 at 12:59 PM MBControl-yyc ***@***.***>
wrote:

> It is a lot of people, but probably not enough to make the league take too
> much notice. With that said, my league has taken 8-10 people that might
> watch their favorite team play each week, now watch 99% of all CFL games
> broadcast (expect for Montreal home games. Horn guy sucks). They know the
> 2nd string back DB's for Ottawa, watching CFL news and rumors.
>
> This type of fantasy encourages a much higher level of engagement to the
> league. But as much as I love CFL football. The league is a very poorly run
> league.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mdryden/110yards/discussions/234#discussioncomment-6125748>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APOMMCQ72B4L5O6IZQTC7VTXKIOI3ANCNFSM6AAAAAAY6DRTP4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
*David Schnare*
Kinetisense Inc. Co-founder
403-526-3556

**Comment by davidkinetisense** [comment]
We looked at this but our group does not want it.  Part of the fun is
waiver and trades.

**Comment by davidkinetisense** [comment]
The CFL site does not evening load right now.

**Comment by mdryden** [comment]
I've created a discord server for 110 yards - if you're interested in brainstorming possible paths forward, or just want to wallow in sadness with me, ~~here's a link that should be good for 7 days~~ (removed link).

Please set your nickname on the server to the same name you're using on this thread so I know who you are.

**Reply by DaCommish** [reply]
Are there any status updates?

**Comment by mdryden** [comment]
Yup just about to post one.
